src/helpers/indexedDB/quiz.ts

```diff
@@ -41,57 +36,60 @@ export const saveQuiz = async ({
 
       return category;
     });
-    await quizzesC.update({ quizId }, { name, categories, quizId, isDraft, userId, numberOfQuestionsPerCategory });
+    await quizzesC.update({ quizId: data.quizId }, data);
   } else {
-    await quizzesC.insert({ name, categories, quizId, isDraft, userId, numberOfQuestionsPerCategory });
+    await quizzesC.insert(data);
   }
+
+  return data;
 };
 
 export const unDraftQuiz = async (quizId: string | number) => {
   await quizzesC.update({ quizId }, { isDraft: false });
 };
 
-export const saveQuestion = async (
-  { QuestionId, Text, Points, TimeLimit, Options }: QuestionParams,
-  quizId: string | number,
-) => {
-  const quiz = (await quizzesC.findOne({ quizId })) as Quiz;
-  const questionData = {
-    questionId: QuestionId,
-    text: Text,
-    points: Points,
-    timeLimit: TimeLimit,
-    options: Options,
-    categoryId: 0,
-  };
+export const saveQuestion = async (questionData: Question, quizId: string | number) => {
+  const quiz = (await quizzesC.findOne({ quizId })) as QuizParams;
   let questionIndex = -1;
   const categoryIndex = quiz.categories.findIndex((category) => {
-    const index = category.questions.findIndex((question) => question.questionId === QuestionId);
+    const index = category.questions.findIndex((question) => question.questionId === questionData.questionId);
 
     if (index >= 0) {
       questionIndex = index;
     }
 
     return category.questions[index];
   });
-
-  const category = quiz.categories[categoryIndex];
+  questionData.options.forEach((option) => {
+    option.optionId = option.optionId || generateRandomNumber();
+  });
 
   if (questionIndex >= 0) {
-    questionData.categoryId = category.categoryId as number;
-    category.questions[questionIndex] = questionData;
+    quiz.categories[categoryIndex].questions[questionIndex] = questionData;
   } else {
-    category.questions.push(questionData);
+    quiz.categories[categoryIndex].questions.push(questionData);
   }
 
-  await quizzesC.update({ quizId }, quiz);
+  await quizzesC.update(
+    { quizId },
+    {
+      categories: quiz.categories,
+      isDraft: quiz.isDraft,
+      name: quiz.name,
+      numberOfQuestionsPerCategory: quiz.numberOfQuestionsPerCategory,
+      quizId: quiz.quizId,
+      userId: quiz.userId,
+    },
+  );
 };
 
 export const getQuizzes = async (userId: number): Promise<Object[]> => {
   const quizzes = await quizzesC.find({ userId }).toArray();
 
   if (quizzes.length > 0) {
     return quizzes;
+  } else if (isGuestUser()) {
```

Comment:
Why are we doing this?

Reply:
I don't want to throw error for guest user, so in parent function it will not go in catch and api call will not trigger